src/librustc_typeck/check/mod.rs

```diff
@@ -2449,6 +2449,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         let mut expected_arg_tys = expected_arg_tys;
         let expected_arg_count = fn_inputs.len();
 
+        let sp_args = if args.len() > 0 {
+            Span { lo: args[0].span.lo, ..args[args.len()-1].span}
```

Comment:
Sadly, we can't safely mix spans like this because some arguments may come from incompatible locations (like from macros).